src/mlpack/tests/image_load_test.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file image_load_test.cpp
+ * @author Mehul Kumar Nirala
+ *
+ * Tests for loading and saving images.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+#include <ctime>    // For time().
+#include <cstdlib>  // For srand().
+#include <boost/test/unit_test.hpp>
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+
+#ifdef HAS_STB // Compile this only if stb is present.
+
+BOOST_AUTO_TEST_SUITE(ImageLoadTest);
+
+/**
+ * Test the invalid extension.
+ */
+BOOST_AUTO_TEST_CASE(LoadInvalidExtensionFile)
+{
+  arma::Mat<unsigned char> matrix;
+  data::ImageInfo info;
+  BOOST_REQUIRE_THROW(data::Load("invalidExtendion.p4ng", matrix, info,
+    false, true),  std::runtime_error);
+}
+
+/**
+ * Test the images is loaded correctly into the matrix using the API.
+ */
+BOOST_AUTO_TEST_CASE(LoadImageAPITest)
+{
+  arma::Mat<unsigned char> matrix;
+  data::ImageInfo info;
+  BOOST_REQUIRE(data::Load("test_image.png", matrix, info, false,
+      true) == true);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, 50 * 50 * 3); // width * height * channels.
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, 1);
+}
+
+/**
+ * Test if the image is saved correctly using API.
+ */
+BOOST_AUTO_TEST_CASE(SaveImageAPITest)
+{
+  data::ImageInfo info;
+  info.Width() = info.Height() = 5;
```

Comment:
Not sure, but using the ``ImageInfo`` constructor might be cleaner, what do you think?

Reply:
Oh, I guess constructor will be good.